### DISCUSSION TITLE
fastdeploy-python部署ppstructurev2怎么处理推理返回的result?

### DISCUSSION THREAD

**Discussion by gl94** [PaddlePaddle/PaddleOCR#12450] [Q&A] [answered]
版本：fastdeploy-python 1.1.2
[使用官方examples代码可以正确推理](https://github.com/PaddlePaddle/FastDeploy/blob/005dc860f2dc06526f06ad83b493e9d5e38160bd/examples/vision/ocr/PP-OCR/cpu-gpu/python/infer_ppstructurev2_table.py)
对于推理返回的result怎么拿到表格识别的结果？
`result = ppstructurev2_table.predict(im)`
result是一个<class 'fastdeploy.libs.fastdeploy_main.vision.OCRResult'>对象
[我在result.h的查看到其定义为：](https://github.com/PaddlePaddle/FastDeploy/blob/005dc860f2dc06526f06ad83b493e9d5e38160bd/fastdeploy/vision/common/result.h)
`struct FASTDEPLOY_DECL OCRResult : public BaseResult {
  std::vector<std::array<int, 8>> boxes;

  std::vector<std::string> text;
  std::vector<float> rec_scores;

  std::vector<float> cls_scores;
  std::vector<int32_t> cls_labels;

  std::vector<std::array<int, 8>> table_boxes;
  std::vector<std::string> table_structure;
  std::string table_html;

  ResultType type = ResultType::OCR;

  void Clear();

  std::string Str();
};`
但是在python中无法用
`result.table_structure`
访问其表格识别结果
最终只能以result.__str__的方式简单粗暴地处理，请我我应该如何处理最终结果呢？

**Comment by happyxhw** [comment, marked as answer]
我更改了 https://github.com/PaddlePaddle/FastDeploy/blob/79a3587339f0552cce22c3dd0066d21db1391db7/fastdeploy/vision/vision_pybind.cc#L157

```
  pybind11::class_<vision::OCRResult>(m, "OCRResult")
      .def(pybind11::init())
      .def_readwrite("boxes", &vision::OCRResult::boxes)
      .def_readwrite("text", &vision::OCRResult::text)
      .def_readwrite("rec_scores", &vision::OCRResult::rec_scores)
      .def_readwrite("cls_scores", &vision::OCRResult::cls_scores)
      .def_readwrite("cls_labels", &vision::OCRResult::cls_labels)
      .def_readwrite("table_structure", &vision::OCRResult::table_structure)
      .def_readwrite("table_html", &vision::OCRResult::table_html)
      .def("__repr__", &vision::OCRResult::Str)
      .def("__str__", &vision::OCRResult::Str);
```

```python
def ocr_to_json(result):
    # bug
    # https://github.com/PaddlePaddle/FastDeploy/issuqes/2429
    # 解决: 重新编译 ./data/python/*.whl
    r_json = {
        "boxes": result.boxes,
        "text": result.text,
        "rec_scores": result.rec_scores,
        "cls_scores": result.cls_scores,
        "cls_labels": result.cls_labels,
        "table_structure": result.table_structure,
        "table_html": result.table_html,
    }
    return json.dumps(r_json)
```

重新编译 fastdeploy

**Reply by tammy-na** [reply]
如何重新编译 fastdeploy？

**Reply by tammy-na** [reply]
c++编译还是python编译就行？没做过编译安装，求指点~

**Reply by happyxhw** [reply]
> c++编译还是python编译就行？没做过编译安装，求指点~

你可以看下官方编译 fastdeploy 流程
